cli/tools/fmt.rs

```diff
@@ -223,23 +278,84 @@ pub fn format_json(
   dprint_plugin_json::format_text(file_path, file_text, &config)
 }
 
-/// Formats a single TS, TSX, JS, JSX, JSONC, JSON, or MD file.
+/// Formats IPYNB files.
+///
+/// attempt to format `code` cells if metadata indicates it's TypeScript.
+fn format_ipynb(
+  ipynb: Ipynb,
+  fmt_options: &FmtOptionsConfig,
+) -> Result<Option<String>, AnyError> {
+  fn source_from(text: String) -> Source {
+    Source::Array(text.split_inclusive('\n').map(|x| x.into()).collect())
+  }
+
+  let is_typescript = ipynb.is_typescript();
+  let cells: Vec<Result<Cell, Cell>> = ipynb
+    .cells
+    .into_iter()
+    .map(|cell| {
+      let source = match cell.source {
+        Source::String(text) => text,
+        Source::Array(texts) => texts.join(""),
+      };
+
+      let formatted = match cell.cell_type.as_str() {
+        "code" if is_typescript => Some("temp.ts"),
+        "markdown" => Some("temp.md"),
+        _ => None,
+      }
+      .map(|path| format_file(&PathBuf::from(path), &source, fmt_options))
+      .unwrap_or(Ok(None));
+
+      match formatted {
+        Ok(Some(text)) => Ok(Cell {
+          source: source_from(text),
+          ..cell
+        }),
+        _ => Err(Cell {
+          source: source_from(source),
+          ..cell
+        }),
+      }
+    })
+    .collect();
+
+  let already_formatted = cells.iter().all(|cell| cell.is_err());
+
+  if already_formatted {
+    return Ok(None);
+  }
+  let cells = cells
+    .into_iter()
+    .map(|cell| cell.unwrap_or_else(|cell| cell))
+    .collect::<Vec<_>>();
+
+  let result = serde_json::to_string_pretty(&Ipynb { cells, ..ipynb });
+
+  Ok(result.map(Some)?)
+}
+
+/// Formats a single TS, TSX, JS, JSX, JSONC, JSON,  MD, or IPYNB file.
 pub fn format_file(
   file_path: &Path,
   file_text: &str,
   fmt_options: &FmtOptionsConfig,
 ) -> Result<Option<String>, AnyError> {
   let ext = get_extension(file_path).unwrap_or_default();
-  if matches!(
-    ext.as_str(),
-    "md" | "mkd" | "mkdn" | "mdwn" | "mdown" | "markdown"
-  ) {
-    format_markdown(file_text, fmt_options)
-  } else if matches!(ext.as_str(), "json" | "jsonc") {
-    format_json(file_path, file_text, fmt_options)
-  } else {
-    let config = get_resolved_typescript_config(fmt_options);
-    dprint_plugin_typescript::format_text(file_path, file_text, &config)
+
+  match ext.as_str() {
+    "md" | "mkd" | "mkdn" | "mdwn" | "mdown" | "markdown" => {
+      format_markdown(file_text, fmt_options)
+    }
+    "json" | "jsonc" => format_json(file_path, file_text, fmt_options),
+    "ipynb" => {
```

Comment:
Will these `ipynb` files be found when just doing `deno fmt`?

Reply:
yes, it was my intention. since markdown files are found automatically, i think it'd be good to do the same for ipynb files.